### PR TITLE
[DS-12131][master] Optimize the log printing of the master module acc…

### LINKED ISSUE #12131: [Improvement][Master] Optimize the log printing of the master module according to the log specification.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The log printing in the current master module has been optimized in #10516, and there are still some other non-standard parts.
According to the log specification in pr https://github.com/apache/dolphinscheduler/pull/11484, I will improve the log printing of the master module.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by guowei-su**
## Purpose of the pull request
This pull request optimizes some log printing of the master module.

## Brief change log

- change some log format  according to the log pecification documentation.
- add some log print.
- close [#12131](https://github.com/apache/dolphinscheduler/issues/12131)
- doc [[Improvement-11481][docs] Add log specification document to contribution guidelines.](https://github.com/apache/dolphinscheduler/pull/11484)

### CODE REVIEW

**Review by brave-lee — DISMISSED** [review]
+1

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/CommonTaskProcessor.java [hunk 1]
```diff
@@ -122,17 +122,17 @@ public boolean dispatchTask() {
...
 
             taskPriority.setTaskExecutionContext(taskExecutionContext);
 
             taskUpdateQueue.put(taskPriority);
-            logger.info("Master submit task to priority queue success, taskInstanceId : {}", taskInstance.getId());
+            logger.info("Task {} is submitted to priority queue success by master", taskInstance.getName());
             return true;
         } catch (Exception e) {
-            logger.error("submit task error", e);
+            logger.info("Task {} is submitted to priority queue error", taskInstance.getName(), e);
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/CommonTaskProcessor.java]:
```suggestion
            logger.error("Task {} is submitted to priority queue error", taskInstance.getName(), e);
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java [hunk 2]
```diff
@@ -159,8 +159,10 @@ private void notifyProcessChanged(ProcessInstance finishProcessInstance) {
             TaskInstance taskInstance = entry.getValue();
             String address = NetUtils.getAddr(masterConfig.getListenPort());
             if (processInstance.getHost().equalsIgnoreCase(address)) {
+                logger.info("Process host is local master, will notify it");
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: add taskInstanceId and workflowId

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java [hunk 3]
```diff
@@ -159,8 +159,10 @@ private void notifyProcessChanged(ProcessInstance finishProcessInstance) {
...
                 this.notifyMyself(processInstance, taskInstance);
             } else {
+                logger.info("Process host is remote master, will notify it");
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java]: same here

**Review by caishunfeng — CHANGES_REQUESTED** [review]

**Comment by caishunfeng** [comment]
Hi @guowei-su the fail CI need to rebase the dev branch.

**Review by brave-lee — APPROVED** [review]
+1

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**